containers-tests/tests/map-properties.hs

```diff
@@ -1365,46 +1450,116 @@ prop_splitModel n ys = length ys > 0 ==>
   in  toAscList l == sort [(k, v) | (k,v) <- xs, k < n] &&
       toAscList r == sort [(k, v) | (k,v) <- xs, k > n]
 
-prop_foldr :: Int -> [(Int, Int)] -> Property
-prop_foldr n ys = length ys > 0 ==>
-  let xs = List.nubBy ((==) `on` fst) ys
-      m  = fromList xs
+prop_fold :: Map Int Int -> Property
+prop_fold m = size m > 0 ==>
+  let xs = toList m
+  in  Foldable.fold (map (:[]) m) == List.map snd xs
+
+prop_foldMap :: Map Int Int -> Property
+prop_foldMap m = size m > 0 ==>
+  let xs = toList m
+  in  Foldable.foldMap (:[]) m == List.map snd xs
+
+prop_foldMapWithKey :: Map Int Int -> Property
+prop_foldMapWithKey m = size m > 0 ==>
+  let xs = toList m
+  in  foldMapWithKey (\k v -> [k,v]) m == concatMap (\(k,v) -> [k,v]) xs
```

Comment:
What is this trying to test? I don't understand. We could do something very simple, like this:

```haskell
prop_foldMapWithKey :: Map Int A -> Property
prop_foldMapWithKey = \m -> foldMapWithKey f m === foldMap (uncurry f) (toList m)
  where
    f k v = [(k, v)]
```